### ISSUE TITLE
implement calculation of hyperparameters from lambdas 

### ISSUE THREAD

**Issue by bdemeshev** [bdemeshev/bvarr#11]

**Comment by AlexSav777** [comment]
И все ж хотелось бы чтоб был реализован расчет гиперпараметров по лямбдам.  Как то скучно без этого.

**Comment by bdemeshev** [comment]
А зачем? Способ с созданием дамми-наблюдений самый робастный по моему опыту. Поэтому сейчас из лямбд напрямую создаются дамми-наблюдения. 

А если очень охота получить гиперпараметры, то можно пройти по цепочке лямбды в дамми-наблюдения, дамми-наблюдения в гиперпараметры, `bvar_conj_lambda2dummy()`, `bvar_conj_dummy2hyper()`.

**Comment by AlexSav777** [comment]
Вот такая конструкция (см. ниже) позволила мне вводить разные значения лямбд и соответственно делать разные прогнозы на выходе.
Несмотря на то что после команды setup<-... выдает сообщение "Calculation of hyperparameters from lambdas is not implemented yet :(  You may safely ignore the message if you supply lambdas :)"
Правомерно ли использовать эти команды таким образом? 

setup <- bvar_conj_setup(z1, p = 4, lambda = c(0.2, 1, 1, 1, 100, 100))
model <- bvar_conj_estimate(setup, keep = 10)
bvar_conj_forecast(model, h = 12, output = "wide") 
(z1 - это мои данные).

А вообще поработал с командами (стандартная конструкция)
setup <- bvar_conj_setup(z1, p = 4) #менял  только лаги
model <- bvar_conj_estimate(setup) 
bvar_conj_forecast(model, h = 12, output = "wide")

И выяснил, что прогноз ИПЦ на 12 месяцев вперед очень близок к тестовой выборке, а еще близок по вневыборочному прогнозу с прогнозом, полученным мной другим методом (множественной регрессией), (в обоих случаях RMSE чуть больше 0,3), что само по себе есть очень хорошо. (Модель из 8 переменных на региональных данных). 
Правда плотно поработать с пакетом еще не успел, но первое впечатление от  BVARR более чем положительное.

**Comment by bdemeshev** [comment]
Как написано в сообщении You may safely ignore the message if you supply
lambdas :)
И в первой команде lambda заданы :) В этом проблем нет!

keep = 10 просто несерьезно, это же монте-карло!
являются ли данные лямбды хорошими для ипц, это только экспериментально
можно понять

ср, 8 янв. 2020 г. в 15:46, AlexSav777 <notifications@github.com>:

**Comment by AlexSav777** [comment]
Спасибо,  большое!

**Comment by AlexSav777** [comment]
Да, хотел еще спросить. Имеется ли возможность в bvarr строить график IRF и Variance Decomposition?

**Comment by bdemeshev** [comment]
если напишешь и закоммитишь, то будет :)  Хотя в целом conjugate normal
inverse wishart — это не про irf :)

Il gio 9 gen 2020, 14:24 AlexSav777 <notifications@github.com> ha scritto:

**Comment by AlexSav777** [comment]
К сожалению у меня еще не тот уровень знаний, чтобы писать. Я пока только пользователь. 
Но впрочем очень жаль, что нет возможности оценить вклад каждого из факторов в прогнозные значения.  Или все таки есть? И просто я не досконально изучил пакет?

**Comment by bdemeshev** [comment]
Моё мнение, что оценка вклада факторов — это ловля мифических единорогов :)
в пакете этого нет.

Il dom 12 gen 2020, 05:32 AlexSav777 <notifications@github.com> ha scritto:
